lib/hardware.go

```diff
@@ -0,0 +1,47 @@
+package lib
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/cybozu-go/well"
+)
+
+func DetectCPUNumber() (int, error) {
+	data, err := os.ReadFile("/proc/cpuinfo")
+	if err != nil {
+		return -1, err
+	}
+	info := strings.Split(string(data), "\n")
+	count := map[string]struct{}{}
+	for _, s := range info {
+		if !strings.Contains(s, "physical id") {
+			continue
+		}
+		if _, ok := count[s]; !ok {
+			count[s] = struct{}{}
+		}
+	}
+	return len(count), nil
+}
+
+func DetectMemoryNumber(ctx context.Context) (int, error) {
+	checkCmd := well.CommandContext(ctx, "dmidecode", "-t", "memory")
+	out, err := checkCmd.Output()
```

Comment:
You can reduce intermediate variables like `checkCmd`, `info`, and `ss`.

```suggestion
	out, err := well.CommandContext(ctx, "dmidecode", "-t", "memory").Output()
```